lib/components/user/begin-save-trip-button.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Button } from 'react-bootstrap'
+
+import { routeTo } from '../../actions/ui'
+
+/**
+ * TODO? Embed _handleClick inside itinerary summary.
+ * This button redirects to /savetrip to let a user define a monitored trip.
+ */
+class BeginSaveTripButton extends Component {
```

Comment:
I think this component is overkill. I think what you're looking for is LinkContainer in 'react-router-bootstrap':

E.g.: 

https://github.com/ibi-group/datatools-ui/blob/master/lib/editor/components/EditorHelpModal.js#L144-L146

Reply:
I have not been successful so far. This code is causing internal errors inside LinkContainer (anything to do with how the router is setup here vs datatools-ui?):
```js
<LinkContainer to='/savetrip'>
    <Button>Save</Button>
</LinkContainer>
```

Error: 
```log
Uncaught TypeError: Cannot read property 'push' of undefined.
```

Reply:
@binh-dam-ibigroup, it looks like the library may not work with react-router v5. Will this work for you? https://reactrouter.com/web/api/Link/component-reactcomponent

Reply:
This is not working either, and is showing a message saying `<Link>` should not be used outside a router. It looks that and the 'push' error are related.

Reply:
OK, then please just make a more generic `link-button.js` file out of this that takes some more generic props and use in those other places I recommended LinkContainer.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/174/commits/9592581707b6610f5a26f3c7992b72171b9b11da